com.unity.multiplayer.mlapi/Runtime/Core/NetworkManager.cs

```diff
@@ -328,6 +328,15 @@ private void Init(bool server)
                 return;
             }
 
+            //This 'if' should never enter
+            if (SnapshotSystem != null)
```

Comment:
I probably missed earlier discussion about this. But why do we have an if which should never enter without logging a warning/error?

Reply:
Defensive programming. `SnapshotSystem` should be null. But if it is not, the next line will be `SnapshotSystem = new SnapshotSystem();` which will skip disposal. Normally skipping disposal would not be a big deal either. It would just happen later at GC time.

But:
- We use a pattern in MLAPI with "strange" pools that require explicit disposal. So, if the `SnapshotSystem` were to be put in such pools, we'd get into issues.
- There also might be code in Dispose() that needs to run right away and not wait 

Lastly, the pattern was copied from `NetworkTickSystem` which had such discussions.